src/frontend/src/lib/utils/qr-code.utils.ts

```diff
@@ -0,0 +1,35 @@
+import type { QrResponse, QrResult } from '$lib/types/qr-code';
+import { decodePayment } from '@dfinity/ledger-icrc';
+import { isNullish, nonNullish, type Token } from '@dfinity/utils';
+
+export const decodeQrCode = ({
+	result,
+	code,
+	requiredToken
+}: {
+	result: QrResult;
```

Comment:
nitpick: `result` feels a bit unconventional maybe for a variable / parameter name?